### PR TITLE
Allow to split the html output by month

### LINKED ISSUE #103: Feature Request: Html file generated from large twitter archive can't be loaded by browser
After processing my archive I ended up with 

4048380 Nov 22 22:30  TweetArchive.html

which chrome refused to load.

An option to split by month or year might be useful.

Note: firefox was able to consume the file

### PR DESCRIPTION
**Pull request by ajuckler**
Add a user prompt asking whether to split the html output by month, the same way as the markdown output is being split.  
The markdown and html filenames use the same pattern.

This closes #103

### CODE REVIEW

File: parser.py [hunk 1]
```diff
@@ -404,26 +404,36 @@ def parse_tweets(input_filenames, username, users, html_template, archive_media_
...
-        markdown_filename = f'{dt.year}-{dt.month:02}-01-Tweet-Archive-{dt.year}-{dt.month:02}.md' # change to group by day or year or timestamp
-        grouped_tweets_markdown[markdown_filename].append(md)
+        filename = f'{dt.year}-{dt.month:02}-01-Tweet-Archive-{dt.year}-{dt.month:02}' # change to group by day or year or timestamp
+        grouped_tweets[filename].append((md, html))
 
-    # Write into *.md files
-    for filename, md in grouped_tweets_markdown.items():
-        md_string =  '\n\n----\n\n'.join(md)
-        with open(filename, 'w', encoding='utf-8') as f:
+    user_input = input(f'Group the html output by month? [y/n]')
```
Comment (timhutton) [inline parser.py:415]: Let's just split by default.

File: parser.py [hunk 2]
```diff
@@ -404,26 +404,36 @@ def parse_tweets(input_filenames, username, users, html_template, archive_media_
...
+    split_html = user_input.lower() in ('y', 'yes')
+
+    for filename, content in grouped_tweets.items():
```
Comment (timhutton) [inline parser.py:418]: `for filename, (md, html) in grouped_tweets.items():`

Comment (ajuckler) [reply parser.py:418]: `grouped_tweets` contains a list of tuples so this wouldn't work.

I can change it into a tuple of lists
```python
grouped_tweets = defaultdict(lambda : (list, list))
for timestamp, md, html in tweets[:2]:
    # [...]
    grouped_tweets[filename][0].append(md)
    grouped_tweets[filename][1].append(html)
```
Or change the loop for something like
`for filename, (md, html) in zip(grouped_tweets.keys(), map(lambda el : zip(*el), grouped_tweets.values())):`

Comment (timhutton) [reply parser.py:418]: Oh sorry, you're quite right.

I think I prefer your original approach but maybe with this style in the joins:

`md, _ for md, html in content`

File: parser.py [hunk 3]
```diff
@@ -404,26 +404,26 @@ def parse_tweets(input_filenames, username, users, html_template, archive_media_
...
-    all_html_string = '<hr>\n'.join(html for _, _, html in tweets)
-    with open(output_html_filename, 'w', encoding='utf-8') as f:
-        f.write(html_template.format(all_html_string))
+        # Write into *.html files
+        html_string = '<hr>\n'.join(html for _, html in content)
+        with open(f'{filename}.html', 'w', encoding='utf-8') as f:
+            f.write(html_template.format(html_string))
 
-    print(f'Wrote {len(tweets)} tweets to *.md and {output_html_filename}, with images and video embedded from {output_media_folder_name}')
+    print(f'Wrote {len(tweets)} tweets to *.md and "*.html", with images and video embedded from {output_media_folder_name}')
```
Comment (timhutton) [inline parser.py:426]: `to *.md and "*.html"` -> `to *.md and *.html`

**Review by timhutton — DISMISSED** [review]

File: parser.py (hunk 3 above)
Comment (ajuckler) [reply parser.py]: Sorry I overlooked this one, fixed

**Review by timhutton — APPROVED** [review]

**Comment by timhutton** [comment]
Thanks @ajuckler! It works well on my archive.

**Comment by martinbrook** [comment]
Thanks @ajuckler , also works great on my large archive

**Pull request merged**